### ISSUE TITLE
Undecryptable messages still occur

### ISSUE THREAD

**Issue by kaepora** [cryptocat/cryptocat#128]
Likely due to storage inconsistencies, reconnection problems... couldn't tell you.

**Comment by bugissue** [comment]
im guessing this refers to _This message could not be decrypted. You may want to ask your buddy to send it again._

was getting this today. it was extremely confusing because i thought i had sent a message that said that and he was copying and pasting it to me so i know that he got that. it was red instead of blue in the background so i should have realized but since it included his username above it, it was still confusing.

**Comment by kaepora** [comment]
This is my highest priority. Currently busy with my actual day job, but my next Cryptocat bug to fix will definitely be this one.

**Comment by Dyras** [comment]
Still happening for me even with #128

**Comment by kaepora** [comment]
Yup, I know.

**Comment by David-R-SF** [comment]
We just started testing. We get the "can't be decrypted" message at the receiving side for anything with Japanese characters in it. (We have a lot of Asian characters in our chat).
This seems to be 100% replicable, and works in either direction. The sender sees the text properly.

**Comment by kaepora** [comment]
@David-R-SF Cannot reproduce:

![screenshot from 2016-06-22 21-57-14](https://cloud.githubusercontent.com/assets/9953/16283345/52687bee-38c4-11e6-95f7-64944ebd30f5.png)

**Comment by cats-meow** [comment]
My partner uses WIN7 and I use Linux Mint 17.3 and we both get the famous "This message could not be decrypted. You may want to ask your buddy to send it again."  Not all the time, but most.  He is on cable modem and I am on 4G hot spot via WiFi.  No problem with other crypto like Jitsi and OTR or Ricochet.  This morning I received 7 CC msgs and none could be decrypted.

**Comment by kaepora** [comment]
@cats-meow Are you both using the latest version of Cryptocat?

**Comment by cats-meow** [comment]
For speedy reply, am answering email, will check GIT later.

Help, Check for updates says:
You are running the latest version of Cryptocat (3.1.26).

regards

**Comment by cats-meow** [comment]
For speedy reply, am answering email, will check GIT later. Help, Check for updates says: You are running the latest version of Cryptocat (3.1.26).

In addition, my partner is also using v 3.1.26

**Comment by cats-meow** [comment]
Some more tests.

My partner and I were both online.  Conversation:
Partner(him)1 Jul., 3:13 pm
hello how are you
me(me)1 Jul., 3:13 pm
i am fine

it took my partner several minutes to be able to select, copy, and paste of what he received:
Partner(him)1 Jul., 3:23 pm
i am fine
goodhealth1(cryptocomp)1 Jul., 3:23 pm
This message could not be decrypted. You may want to ask your buddy to send it again.

In other words, he received my reply and immediately afterwards received the famous can not decrypt msg, which in this particular test was a false msg.

Each of us then went off-line in turn and sent a msg while offline.  No problems, everything OK.

**Comment by cats-meow** [comment]
My partner and I can now chat with Cryptocat 3.2.02 and the disturbing error msgs
"can't be decrypted" no longer occur.  Operation appears to be as it should be.

**Comment by Deleetdk** [comment]
This error occurs frequently for me.
![cryptocat fail](https://cloud.githubusercontent.com/assets/6759913/18650116/8ea46b82-7ec3-11e6-8e44-fb6072a72d73.png)

**Comment by Mc128k** [comment]
Here too, OSX to Win10

**Comment by kaepora** [comment]
Are you all using the latest version of Cryptocat, and is your communicating party also using the latest version?

**Comment by Mc128k** [comment]
Just downloaded on both computers when I commented

**Comment by Deleetdk** [comment]
@kaepora Yes. Just downloading the latest versions. This error only happens with one of my contacts, but it happens very frequently. He is using Windows 8.1 too I think.

**Comment by kaepora** [comment]
This is frustrating as I simply cannot reproduce it. Will investigate more when I have time.

**Comment by cats-meow** [comment]
Hi,
I have with the newest CC version begun receiving this error msg again.  
Not so often.
Maybe two times only.  The last time, I received a batch of 5 msgs, and 
only the last one
was the error msg.  the first 4 msgs were ok.

Earl

**Comment by cats-meow** [comment]
Today 05 Oct 2016, version 3.2.07
Of the 7 msgs received, the last 6 could not be decrypted.  Only the 
first was decrypted.

**Comment by Dyras** [comment]
This just happened once for me. Latest Cryptocat on Windows.

**Comment by cats-meow** [comment]
Today with v3.2.07 on Linux MINT 17.3, CC said my partner was typing, 
the blue bar varied in intensity, and then the msg appeared could not be 
decrypted.  This happened for 4 msgs, one after the other.  At the 
moment, CC is unusable. Partner uses WIN 7.

**Comment by konradreiche** [comment]
Please reopen, this is still an issue. Experiencing it with `Linux <-> Windows 10` setup on the latest Cryptocat version.

**Comment by cats-meow** [comment]
Konrad, I am also continuing to have this issue between Linux and Win 7.
At one time, I thought this problem was gone, but no - it continues.
Unable to see why this randomly happens.
I understand that finding the problem is difficult and frustrating for 
the developers.
I hope the problem is found and solved so that CryptoCat can be used for 
more than testing.

**Comment by a9e78rauijasdf** [comment]
I also have the problem. Im on 3.2.07 on win8 with a contact on ubuntu (3.2.07 as well). appears randomly. cant seem to find a pattern in the errors.

**Comment by nklose** [comment]
Please reopen. I frequently get this error with Win10 to Win10 with 3.2.08.

**Comment by Lanzat2** [comment]
This bug hit me heavy, no problem with Linux to Linux chat, but with Linux to Windows almost 70% or message are not  decrypted.

**Comment by billkasper** [comment]
July 8, 2017, happens constantly using latest version with Win 7 64bit on both ends

**Comment by shamranofalive** [comment]
Also the same here and it's keeping me from using Cryptocat :-(

**Comment by ghost** [comment]
Seems to happen randomly, makes the software unusable. Running Fedora 27 64bit on my end. Other people use other OS's. Latest version of Cryptocat on all parties machines.

**Comment by shamranofalive** [comment]
Dammit.. Had to change to Threema bc this error... Please fix

**Comment by plan-a-hamburg** [comment]
I am also experiencing this frequently, but only with the first messages. Once both parties have written something, the error disappears.

**Comment by Cubytus** [comment]
It didn't happen until a few days, but Mac OS X to Windows 10, occurs every two messages. 
Both computers use the same version of Cryptocat, the latest as of this post (3.2.08).

Makes the messenger unusable. We tried logout-login, I added a VPN connection just in case it was a MITM attack, to no avail.
